numba/tests/test_stencils.py

```diff
@@ -365,12 +407,1272 @@ def test_impl(n):
+        """2 args more complexity"""
+        def kernel(a, b):
+            return a[0, 1] + a[1, 2] + b[-2, 0] + b[0, -1]
+        a = np.arange(30.).reshape(5, 6)
+        b = np.arange(30.).reshape(5, 6)
+        self.check(kernel, a, b)
+
+    # doesn't match pyStencil, probably expected
+    def test_basic44(self):
+        """2 args, has assignment before use"""
+        def kernel(a, b):
+            a[0, 1] = 12
+            return a[0, 1]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    # doesn't match pyStencil, probably expected
+    def test_basic45(self):
+        """2 args, has assignment and then cross dependency"""
+        def kernel(a, b):
+            a[0, 1] = 12
+            return a[0, 1] + a[1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    # doesn't match pyStencil, probably expected
+    def test_basic46(self):
+        """2 args, has cross relidx assignment"""
+        def kernel(a, b):
+            a[0, 1] = b[1, 2]
+            return a[0, 1] + a[1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, expected_exception=[ValueError, LoweringError])
+
+    def test_basic47(self):
+        """3 args"""
+        def kernel(a, b, c):
+            return a[0, 1] + b[1, 0] + c[-1, 0]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        c = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, c)
+
+    # matches pyStencil, but all ought to fail
+    # probably hard to detect?
+    def test_basic48(self):
+        """2 args, has assignment before use via memory alias"""
+        def kernel(a):
+            c = a.T
+            c[:, :] = 10
+            return a[0, 1]
+        a = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a)
+
+
+    def test_basic49(self):
+        """2 args, standard_indexing on second"""
+        def kernel(a, b):
+            return a[0, 1] + b[0, 3]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'})
+
+    # this should raise as b[0, 15] is out of bounds for 'b' under standard_indexing
+    def test_basic50(self):
+        """2 args, standard_indexing OOB"""
+        def kernel(a, b):
+            return a[0, 1] + b[0, 15]
+        a = np.arange(12.).reshape(3, 4)
+        b = np.arange(12.).reshape(3, 4)
+        self.check(kernel, a, b, options={'standard_indexing': 'b'}, expected_exception=IndexError)
+
+    # not sure this makes sense, what size array should be returned when both args are labelled as
+    # operating under standard_indexing? It probably ought to raise
+    def test_basic51(self):
```

Comment:
As per the comment, not sure this makes sense, what size array should be returned when both args are labelled as
operating under standard_indexing? It probably ought to raise?

Reply:
Raising now if first argument is standard indexed.

Reply:
Ok, thanks, I think this is a suitable fix. Please could it be documented that `standard_indexing` cannot be applied to the first argument? Thanks.